src/java.desktop/share/classes/sun/swing/FilePane.java

```diff
@@ -1317,13 +1317,6 @@ public void tableChanged(TableModelEvent e) {
             detailsTable.addFocusListener(repaintListener);
         }
 
-        // TAB/SHIFT-TAB should transfer focus and ENTER should select an item.
-        // We don't want them to navigate within the table
-        ActionMap am = SwingUtilities.getUIActionMap(detailsTable);
-        am.remove("selectNextRowCell");
-        am.remove("selectPreviousRowCell");
-        am.remove("selectNextColumnCell");
-        am.remove("selectPreviousColumnCell");
         detailsTable.setFocusTraversalKeys(KeyboardFocusManager.FORWARD_TRAVERSAL_KEYS,
```

Comment:
If I have this right, JTable gets the map initialised via this code in BasicTableUI.java
    InputMap getInputMap(int condition) {
        if (condition == JComponent.WHEN_ANCESTOR_OF_FOCUSED_COMPONENT) {
            InputMap keyMap =
                (InputMap)DefaultLookup.get(table, this,
                                            "Table.ancestorInputMap");
            InputMap rtlKeyMap;

            if (table.getComponentOrientation().isLeftToRight() ||
                ((rtlKeyMap = (InputMap)DefaultLookup.get(table, this,
                                            "Table.ancestorInputMap.RightToLeft")) == null)) {
                return keyMap;
            } else {
                rtlKeyMap.setParent(keyMap);
                return rtlKeyMap;
            }
        }
        return null;
    }

This uses the shared defaults for the L&F. I imagine that sharing is by far the most efficient thing to do for 99%  of uses
, in which case I expect that somewhere there's some doc telling apps that if they mess with the map for a component, they need to make a copy first ? But how to do that ? And the author of the code above that modified the shared map  presumably was ignorant of this.

Reply:
Yes, modifying the shared ActionMap is causing this issue though. As far as I have seen the copy first solution is mentioned in this bug https://bugs.openjdk.org/browse/JDK-8166352 as customer submitted workaround.

Reply:
I did thought of few solutions for this issue:
1. To reset (If possible, but not sure how to do this, yet we have `SwingUtilities.replaceUIActionMap`) the ActionMap. But when to reset is again a question?
2. To consider what customer has suggested about making a copy and then using that which again I'm not sure since here shared defaults are used from BasicTableUI.
3. To remove the lines causing issue which I have proposed. I feel it is safe now to remove it since TAB/ENTER functionalities (Basically TAB being moved out of FilePane and ENTER on selecting file/opening Directory) is handled without these lines too. I did CI test for any regression, but its look fine without this lines too.

Reply:
TAB navigation is prevented by this fix https://bugs.openjdk.org/browse/JDK-4835633 where the TAB is rejected in [this](https://github.com/openjdk/jdk/blame/a81e1bf1e1a6f00280b9be987c03fe20915fd52c/src/java.desktop/share/classes/javax/swing/plaf/basic/BasicTableUI.java#L683) line.